File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/auth/verify/VerifyModule.kt
package au.com.crazybean.mobilex.kurir.modules.auth.verify

import au.com.crazybean.mobilex.kurir.extension.qualifier
import au.com.crazybean.mobilex.kurir.modules.base.ViewModel
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.dsl.module

val verifyModule = module {
    val qualifier = qualifier<VerifyWrapper>()

    // Wrapper via ViewModel
    viewModel(qualifier) {
        ViewModel(VerifyWrapper(get()))
    }

    // Actor
    factory { (scene: VerifyScene?) ->
        val viewModel = get<ViewModel<VerifyWrapper>>(qualifier)
        VerifyActor(scene, viewModel.wrapper)
    }
}

File: ios/Kurir/modules/auth/verify/VerifyModule.swift
//
//  VerifyModule.swift
//  Kurir
//
//  Created by Loren on 9/9/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Foundation
import Mobilex

class VerifyModule: Module {
    override func inject() {
        // Wrapper
        factory(VerifyWrapper.self) { r in
            VerifyWrapper(repository: r.resolve(UsersRepository.self)!)
        }
        
        // Actor
        factory(VerifyActor.self) { (r, scene: VerifyScene) -> VerifyActor in
            VerifyActor(scene: scene, wrapper: r.resolve(VerifyWrapper.self)!)
        }
    }
}
